ctest-next/src/ast/structure.rs

```diff
@@ -15,4 +13,9 @@ impl Struct {
     pub fn ident(&self) -> &str {
         &self.ident
     }
+
+    /// Return the public fields of the struct.
+    pub fn public_fields(&self) -> Vec<&Field> {
```

Comment:
```suggestion
    pub fn public_fields(&self) -> impl Iterator<Item = &Field> {
```

That way you don't need to `.collect()`